libbeat/processors/add_process_metadata/gosigar_cid_provider.go

```diff
@@ -106,25 +107,35 @@ func (p gosigarCidProvider) getProcessCgroups(pid int) (map[string]string, error
 // ID is found then an empty string is returned.
 // Example:
 // /kubepods/besteffort/pod9b9e44c2-00fd-11ea-95e9-080027421ddf/2bb9fd4de339e5d4f094e78bb87636004acfe53f5668104addc761fe4a93588e
-func (p gosigarCidProvider) getCid(cgroups map[string]string) string {
+func (p gosigarCidProvider) getCid(cgroups map[string]cgroup.ControllerPath) string {
 	// if regex defined use it to find cid
 	if len(p.cgroupRegex) != 0 {
 		re := regexp.MustCompile(p.cgroupRegex)
 		for _, path := range cgroups {
-			rs := re.FindStringSubmatch(path)
+			rs := re.FindStringSubmatch(path.ControllerPath)
 			if rs != nil {
 				return rs[1]
 			}
 		}
 	} else {
-		// use string prefix to find cid
+		// else, fall back to a hardcoded regex
+		// In an attempt to not break the user-facing config interface for this processor,
+		// fall back to the config'ed prefixes if we have cgv1, otherwise use regex
+		// This should work with k8s on cgroupsV2, as we're still trying to extract the same container ID
```

Comment:
+1, but it'd be good to confirm that.

Reply:
Yah, I've been struggling to find a prod-like k8s environment that uses V2 for me to test on.

Reply:
Well, if this is difficult to find maybe this is not so important to cover by now.
I'd suggest to add this specific case to the test plan, @ChrsMark or @MichaelKatsoulis may have more ideas about where a k8s environment with cgroups v2 can be found.